test/langtools/tools/javac/T8271079.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8271079
+ * @summary JavaFileObject#toUri in MR-JAR returns real path
+ * @modules java.compiler
+ *          jdk.compiler
+ * @run main T8271079
+ */
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import java.util.*;
+import java.util.jar.JarEntry;
+import javax.tools.*;
+
+public class T8271079 {
+
+  public static void main(String[] args) throws Exception {
+    Path mr = generateMultiReleaseJar();
+    try {
+      testT8271079(mr);
+    } finally {
+      Files.deleteIfExists(mr);
+    }
+  }
```

Comment:
As a matter of style, it is common practice to create an instance of the class in `main` to avoid having an excess of `static` methods.